framework/encode/dx12_state_writer.cpp

```diff
@@ -1089,8 +1088,13 @@ void Dx12StateWriter::WriteCommandListCommands(const ID3D12CommandList_Wrapper*
     size_t         data_size = list_info->command_data.GetDataSize();
     const uint8_t* data      = list_info->command_data.GetData();
 
-    // TODO: Don't write any commands, including the Reset or Close commands, if the command allocator used in the most
-    // recent Reset command no longer exists.
+    bool invalid_command_allocator =
+        (list_info->reset_command_allocator_id != format::kNullHandleId) &&
+        (state_table.GetID3D12CommandAllocator_Wrapper(list_info->reset_command_allocator_id) == nullptr);
+    if (invalid_command_allocator == true)
+    {
+        return;
+    }
```

Comment:
Can this block be moved up above line 1084 `bool write_commands = CheckCommandListObjects(list_info.get(), state_table);`? That call to `CheckCommandListObjects` isn't needed if we are skipping all commands.